### PR TITLE
batch operator script - fix long term errors

### PR DESCRIPTION
**Pull request by edcdavid**
This adding several fixes with the intent for improve stability of the script: 
- install all namespace operators in their own namespace ("test-namespace") instead of openshift-operators. the namespace is deleted after each operator test
- keep installing operators with specific namespaces in their required namespace
- added retry function to wait until oc command succeed
- added function to wait until the cluster becomes reachable

### CODE REVIEW

File: run-basic-batch-operators-test.sh [hunk 1]
```diff
@@ -64,32 +64,113 @@ cleanup() {
...
+		echo "Retry $retries/$max_retries: Waiting for a few seconds before the next attempt..." >&2
+		sleep "$timeout"
+	done
+
+	echo "Maximum retries reached. Exiting with failure." >&2
+	return 1
+}
+
+waitClusterOk() {
+	timeoutSeconds=600
```
Comment (rdavid) [inline run-basic-batch-operators-test.sh:146]: Use local variables. Do not mix camel and snake cases in naming. I vote for snake case everywhere

File: run-basic-batch-operators-test.sh [hunk 2]
```diff
@@ -64,32 +64,113 @@ cleanup() {
...
+					# shellcheck disable=SC2030
+					stdout=$(cat)
+					typeset -p stdout
+				)
+			# shellcheck disable=SC2030
+			status=$?
+			typeset -p status
+		)"
+		# shellcheck disable=SC2031
+		if [ "$status" == 0 ]; then
```
Comment (rdavid) [inline run-basic-batch-operators-test.sh:116]: `==` is not POSIX.
```suggestion
		if [ "$status" -eq 0 ]; then
```

File: run-basic-batch-operators-test.sh [hunk 3]
```diff
@@ -64,32 +64,113 @@ cleanup() {
...
+function withRetry() {
+	local max_retries=$1
+	local timeout=$2
+	shift 2
+
+	local retries=0
+	local status=0
+	until [ "$retries" -ge "$max_retries" ]; do
+		# Execute oc command saving stdout, stderr and exit status
+		# see: https://stackoverflow.com/questions/11027679/capture-stdout-and-stderr-into-different-variables/41069638#41069638
```
Comment (rdavid) [inline run-basic-batch-operators-test.sh:94]: Use short link.
```suggestion
		# see: https://stackoverflow.com/questions/11027679
```

File: run-basic-batch-operators-test.sh (hunk 2 above)
Comment (edcdavid) [reply run-basic-batch-operators-test.sh]: fixed

File: run-basic-batch-operators-test.sh (hunk 1 above)
Comment (edcdavid) [reply run-basic-batch-operators-test.sh:148]: fixed

File: run-basic-batch-operators-test.sh [hunk 4]
```diff
@@ -64,32 +64,121 @@ cleanup() {
...
+
+waitClusterOk() {
+	timeoutSeconds=600
+	startTime=$(date +%s)
+	while true; do
+		status=0
+		oc get nodes &>/dev/null || status="$?"
+		if [ "$status" -eq 0 ]; then
+			return 0
+		fi
```
Comment (rdavid) [inline run-basic-batch-operators-test.sh:155]:
```suggestion
		oc get nodes &>/dev/null || {
			echo >&2 "get nodes failed with error $?."
			return 0
		}
```

File: run-basic-batch-operators-test.sh [hunk 5]
```diff
@@ -64,32 +64,121 @@ cleanup() {
...
+		sleep "$timeout"
+	done
+
+	echo "Maximum retries reached. Exiting with failure." >&2
+	return 1
+}
+
+waitClusterOk() {
+	timeoutSeconds=600
+	startTime=$(date +%s)
```
Comment (rdavid) [inline run-basic-batch-operators-test.sh:149]:
```suggestion
waitClusterOk() {
	local \
		startTime \
		status \
		timeoutSeconds=600
	startTime=$(date +%s 2>&1) || {
		echo >&2 "date failed with error $?: $startTime"
		return 0
	}
```

**Review by rdavid — APPROVED** [review]

File: run-basic-batch-operators-test.sh (hunk 4 above)
Comment (rdavid) [reply run-basic-batch-operators-test.sh:155]: You’re right:
```
	oc get nodes &>/dev/null && return 0
	echo >&2 "get nodes failed with error $?.
```

Comment (edcdavid) [reply run-basic-batch-operators-test.sh]: fixed

File: run-basic-batch-operators-test.sh [hunk 6]
```diff
@@ -64,32 +64,121 @@ cleanup() {
...
+# Executes command with retry
+withRetry() {
+	local maxRetries=$1
+	local timeout=$2
+	shift 2
+
+	local retries=0
+	local status=0
+	local stderr=""
+	local stdout=""
```
[the remark this answers is not in the document]
Comment (edcdavid) [reply run-basic-batch-operators-test.sh]: fixed

File: run-basic-batch-operators-test.sh (hunk 5 above)
Comment (edcdavid) [reply run-basic-batch-operators-test.sh]: fixed

**Review by greyerof — APPROVED** [review]

**Pull request merged**